android/titanium/libv8-services.js

```diff
@@ -42,6 +46,81 @@ async function loadPackageJson() {
 	return fs.readJson('../package.json');
 }
 
+/**
+ * Debug snapshot generation locally.
+ * @param {String} v8SnapshotHeaderFilePath The path to save generated snapshot header.
+ * @param {String} rollupFileContent Javascript content to store in snapshot.
+ */
+async function debugGenerateSnapshot(v8SnapshotHeaderFilePath, rollupFileContent) {
+
+	// Generate startup.js
+	const STARTUP_PATH = path.join(__dirname, '../../dist/tmp', 'startup.js');
+	await fs.writeFile(STARTUP_PATH, `this._startSnapshot = global => { ${rollupFileContent} };`);
+
+	const packageJsonData = await loadPackageJson();
+	const v8TargetVersion = packageJsonData.v8.version;
+	const v8TargetMode = packageJsonData.v8.mode;
+	const V8_LIB_DIRECTORY = path.join(__dirname, '../../dist/android/libv8', v8TargetVersion, v8TargetMode, 'libs');
+
+	// Obtain V8 architectures.
+	const archs = (await fs.readdir(V8_LIB_DIRECTORY, { withFileTypes: true }))
+		.filter(entry => entry.isDirectory())
+		.map(entry => entry.name);
+
+	console.log(`Found architectures ${archs.join(', ')}`);
+
+	let blobs = {};
+
+	// Generate snapshots.
+	for (const arch of archs) {
+		const MKSNAPSHOT_PATH = path.join(V8_LIB_DIRECTORY, arch, 'mksnapshot');
+		const BLOB_PATH = path.join(V8_LIB_DIRECTORY, arch, 'blob.bin');
+		const EMBEDDED_PATH = path.join(V8_LIB_DIRECTORY, arch, 'embedded.S');
+
+		// Delete existing snapshot blob.
+		try {
+			await fs.unlink(BLOB_PATH);
+		} catch (e) {
+			// Do nothing...
+		}
+
+		const args = [
+			'--startup_blob=' + BLOB_PATH,
+			STARTUP_PATH
+		];
+
+		console.log(`Generating snapshot blob for ${arch}...`);
+
+		// Include embedded blob.
+		if (await fs.exists(EMBEDDED_PATH)) {
+			args.unshift(
+				'--turbo_instruction_scheduling',
+				'--embedded_src=' + EMBEDDED_PATH,
```

Comment:
Appending `EMBEDDED_PATH` here will likely create a `\` escaping problem on Windows. We'll need to escape the backslashes. Also, we may need to double quote the path in case it contains spaces.
```javascript
'--embedded_src=' + quotePath(EMBEDDED_PATH.replace(/\\/g, '\\\\')),
```

Reply:
I don't think we need to care about this as the `mksnapshot` executable can't be run on Windows